source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -290,56 +326,28 @@ ComponentInterface<NodeT>::ComponentInterface(
 }
 
 template<class NodeT>
-void ComponentInterface<NodeT>::step() {
-  for (const auto& predicate: this->predicates_) {
-    std_msgs::msg::Bool msg;
-    msg.data = this->get_predicate(predicate.first);
-    auto predicate_iterator = this->predicate_publishers_.find(predicate.first);
-    if (predicate_iterator == this->predicate_publishers_.end()) {
-      RCLCPP_ERROR_STREAM_THROTTLE(this->get_logger(), *this->get_clock(), 10000,
-                                   "No publisher for predicate " << predicate.first << " found.");
-      return;
-    }
-    predicate_publishers_.at(predicate.first)->publish(msg);
-  }
-}
-
-template<class NodeT>
-std::string ComponentInterface<NodeT>::generate_predicate_topic(const std::string& predicate_name) const {
-  return "/predicates/" + std::string(this->get_name()) + "/" + predicate_name;
-}
-
-template<class NodeT>
-void ComponentInterface<NodeT>::add_variant_predicate(
-    const std::string& name, const utilities::PredicateVariant& predicate
-) {
-  if (this->predicates_.find(name) != this->predicates_.end()) {
-    RCLCPP_DEBUG_STREAM(this->get_logger(), "Predicate " << name << " already exists, overwriting.");
-  } else {
-    this->predicate_publishers_.insert_or_assign(
-        name, this->template create_publisher<std_msgs::msg::Bool>(
-            this->generate_predicate_topic(name), 10
-        ));
-  }
-  this->predicates_.insert_or_assign(name, predicate);
+inline void ComponentInterface<NodeT>::step() {
+  this->publish_predicates();
```

Comment:
As we just discussed, this is not correct. For a lifecycle component, publishing should only happen if the component is active

Reply:
Should be good now